janusgraph-backend-testutils/src/main/java/org/janusgraph/testutil/JanusGraphAssert.java

```diff
@@ -81,4 +91,57 @@ private static boolean isEmpty(Object obj) {
         throw new IllegalArgumentException("Cannot determine size of: " + obj);
     }
 
+    public static void assertNumStep(int expectedResults, int expectedSteps, GraphTraversal traversal, Class<? extends Step>... expectedStepTypes) {
+        int num = 0;
+        while (traversal.hasNext()) {
+            traversal.next();
+            num++;
+        }
+
+        assertEquals(expectedResults, num);
```

Comment:
Do we have to iterate and count? What about this:
```suggestion
        assertEquals(expectedResults, traversal.toList().size());
```